[meloppeitreet] iP

meloppeitreet opened the pull request:
# EchoLex
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

EchoLex frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

meloppeitreet committed:
Override Duke.java with EchoLex.java

meloppeitreet committed:
Feature: Add input echo to EchoLex.java

meloppeitreet committed:
Feature: Add list memory command to EchoLex.java

meloppeitreet committed:
Feature: Add task marking to EchoLex.java

meloppeitreet committed:
Reorganize mark command switch cases to method

meloppeitreet committed:
Feature: Add todos, events, and deadlines

meloppeitreet committed:
Add text-ui-test and fix wording

meloppeitreet committed:
Add error handling in EchoLexException.java

meloppeitreet committed:
Add delete functionality to EchoLex.java

meloppeitreet committed:
Add task importance level enum to Task.java

kimseunghyun-kr commented on src/main/java/EchoLex.java:
```diff
@@ -0,0 +1,231 @@
+     * @param description Task description.
+     * @param options Deadline time or event duration.
+     * @return Task created message.
+     */
```
1. Shouldn't the ```throws EchoLexException``` be present within the documentation as a @throws within the javadocs.

2. I like how short the description is, but i think it should be slightly longer, defining what are Tasks

kimseunghyun-kr commented on src/main/java/EchoLexException.java:
```diff
@@ -0,0 +1,13 @@
+public class EchoLexException extends Exception {
```
1. I do not like it that you just extended Exception, this is generally considered bad practice in JAVA, particularly due to the consistent need of defining throws XXXException each time it is used, which quickly piles on in the throws section. Instead, i recommend using RuntimeException, which would be silent

2. I do not like it that there are no javadocs present particularly for Exception related classes as it makes it difficult to identify what kind of exceptional circumstance this is

meloppeitreet committed:
Add save and load functionality to EchoLex.java

meloppeitreet committed:
Merge branch 'branch-Level-7'

meloppeitreet committed:
Add date parsing functionality to EchoLex.java

meloppeitreet committed:
Merge branch 'branch-Level-8'

meloppeitreet committed:
Refactor EchoLex.java structure for more OOP

meloppeitreet committed:
Merge branch 'A-MoreOOP'

meloppeitreet committed:
Organize src/main/java*.java into packages

meloppeitreet committed:
Revert "Organize src/main/java*.java into packages"

This reverts commit 89e2f36d076040b1b937e90e166224aeec582323.

meloppeitreet committed:
Organize src/main/java/*.java into packages

meloppeitreet committed:
Revert "Organize src/main/java/*.java into packages"

This reverts commit 045642dcfe8676ed5b13078a2963b496216afb33.

meloppeitreet committed:
Merge branch 'master' into A-Packages

meloppeitreet committed:
Merge tag 'A-Packages'

meloppeitreet committed:
Revert "Merge branch 'master' into A-Packages"

This reverts commit 96cfb1505db2cf43be15eb6cab8ba86e9d33ab62, reversing
changes made to 89e2f36d076040b1b937e90e166224aeec582323.

Melvan-Chan commented on src/main/java/echolex/command/Command.java:
```diff
@@ -0,0 +1,191 @@
+
+    private final String command;
+    private final String argument;
+    private final HashMap<String, String> options;
```
Good usage of final for these variables to ensure immutability.

Melvan-Chan commented on src/main/java/echolex/command/Command.java:
```diff
@@ -0,0 +1,191 @@
+     * @return Formatted memory list string.
+     */
+    public boolean isExit() {
+        return command.equals("bye");
```
Consider using equalsIgnoreCase() for case insensitivity.

Melvan-Chan commented on src/main/java/echolex/command/Command.java:
```diff
@@ -0,0 +1,191 @@
+        int counter = 1;
+        String result = "Here are the tasks in your list:\n";
+
+        for (Task input : tasks) {
```
Consider handling potential NullPointerException in case tasks is not initialized since tasks are modified in other methods.

meloppeitreet committed:
Merge branch 'add-gradle-support'

Melvan-Chan commented on src/main/java/echolex/command/Parser.java:
```diff
@@ -0,0 +1,76 @@
+        String command = main[0];
+        String argument = "";
+        if (main.length > 1) {
+            argument = main[1];
```
Consider using trim() on main[1] to remove unnecessary spaces.

meloppeitreet committed:
Setup Gradle for EchoLex

meloppeitreet committed:
Add tests and fixes to command and task packages

meloppeitreet committed:
Add JavaDoc headers to Command, Parser, Storage classes

meloppeitreet committed:
Merge branch 'branch-A-JavaDoc'

meloppeitreet committed:
Fix coding standards according to CS2103 conventions

meloppeitreet committed:
Merge branch 'branch-A-CodingStandard'

meloppeitreet committed:
Add find task functionality and tests to Command.java

meloppeitreet committed:
Merge branch 'branch-Level-9'

meloppeitreet committed:
Add support for CheckStyle for finding code style violations

meloppeitreet committed:
Merge branch 'A-CheckStyle'

meloppeitreet committed:
Add basic GUI for EchoLex using JavaFX

meloppeitreet committed:
Set dimensions of EchoLex

meloppeitreet committed:
Merge branch 'Level-10'

meloppeitreet committed:
Add varargs to boxInput in Ui.java

meloppeitreet committed:
Add basic GUI enhancements using JavaFX CSS

meloppeitreet committed:
Merge branch 'branch-Level-10'

meloppeitreet committed:
Modify EchoLex GUI for shifting away from Duke

meloppeitreet committed:
Merge branch 'branch-Level-10'

kimseunghyun-kr commented on src/main/java/echolex/command/Command.java:
```diff
@@ -0,0 +1,253 @@
+
+import echolex.error.EchoLexException;
+import echolex.task.Deadline;
+import echolex.task.Event;
```
1. I like that you followed consistent ordering of import statements. very nice

kimseunghyun-kr commented on src/main/java/echolex/command/Command.java:
```diff
@@ -0,0 +1,253 @@
+    public String execute(TaskList tasks) {
+
+        switch (command) {
+        case "list":
```
1. i like that you followed switch coding style
2. I would like you to consider using lambda style switch statements to make your code more readable

kimseunghyun-kr commented on src/main/java/echolex/command/Ui.java:
```diff
@@ -0,0 +1,82 @@
+public class Ui {
+
+    /** Chat Indent */
+    static final String INDENT = "    "; // 4 spaces
+    /** Chat Delimiter: Horizontal Line */
+    static final String HORIZONTAL_LINE = "____________________________________________________________";
+
+    /** Chat Hello / Goodbye */
+    static final String HELLO = "Hello! I'm EchoLex" + "\n" + "What can I do for you?";
+    static final String GOODBYE = "Bye. Hope to see you again soon!";
+
```
1. i like that you followed coding standard to use fully uppercase for constants.

kimseunghyun-kr reviewed (commented):
Overall, LGTM. The code was very readable and follows java coding standards. OOP principles were well followed with suitable inheritance being shown. Good job! However, i hope you include more javadocs in the future, which makes the code more understandable.

meloppeitreet committed:
Remove original Duke images

meloppeitreet committed:
Add assert statements for documentation of assumptions

The new assert statements helps document assumptions within the code,
making it easier to catch unexpected behavior during development.

These assertions act as inline documentation, clarifying expected
conditions and preventing silent failures in areas with heavy logic:

* Command.java
* Parser.java
* Storage.java

meloppeitreet committed:
Improve code quality for readability and safe practices

* Reduced long methods
* Use the default branch
* Minimized code duplication

meloppeitreet committed:
Add stream to find command in Command.java

meloppeitreet committed:
Merge pull request #2 from meloppeitreet/branch-A-CodeQuality

Improve code quality for readability and safe practices

meloppeitreet committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/echolex/command/Command.java
#	src/main/java/echolex/command/Storage.java

meloppeitreet committed:
Merge pull request #1 from meloppeitreet/branch-A-Assertions

Add assert statements for documentation of assumptions

meloppeitreet committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/echolex/command/Command.java

meloppeitreet committed:
Merge pull request #3 from meloppeitreet/branch-A-Streams

Add stream to find command in Command.java

meloppeitreet committed:
Refactor Command class into subclasses and reorganize packages

For improving code maintainability and separation of concerns

* Split to subclasses to handle different command types.
* Combined subclasses and Command class into command package.
* Moved other classes into a new utility package.

meloppeitreet committed:
Set Up GitHub Action for No-Op Workflow

GitHub Actions should recognise the workflow
but do nothing when a push occurs.

This is purely for understanding purposes.

meloppeitreet committed:
Fix GitHub Action for No-Op Workflow

meloppeitreet committed:
Add schedule viewing functionality to EchoLex.java

- Implemented "schedule" command.
- Displays summary schedule of upcoming events and deadlines.
- Added "schedule /on <date>" command to view a specific date.
- Included handling for cases with no events or deadlines.

meloppeitreet committed:
Merge pull request #4 from meloppeitreet/B-ViewSchedules

Add schedule viewing functionality to EchoLex.java

meloppeitreet committed:
Improve GUI for dialog boxes and window title with icon

* User dialog box has same colour as command text box
* Window has a title and icon
* EchoLex dialog box has a shadow

meloppeitreet committed:
Merge branch 'A-BetterGui'

meloppeitreet committed:
Modify welcome and goodbye messages for EchoLex

Add some personality to EchoLex to distinguish slightly from Duke

meloppeitreet committed:
Modify welcome and goodbye messages for EchoLex

Format the message and change it a little more

meloppeitreet committed:
Merge branch 'A-Personality'

meloppeitreet committed:
Handle errors for non-number user input expecting index

Applies to delete and mark commands

meloppeitreet committed:
Add user guide and JavaDoc comments

Clean up of product with README.md description and code comments
